functions/go/apply-replacements/replacements/replacements.go

```diff
@@ -43,11 +42,7 @@ func (r *Replacements) Process(rl *fn.ResourceList) error {
 	}
 	transformedItems, err := r.Transform(rl.Items)
 	if err != nil {
-		rl.Results = append(rl.Results, &fn.Result{
```

Comment:
I'm using `fn.AsMain` which already does this step, so we don't need to do it here.

Reply:
I commented in the other PR. Does it make sense to change the `Process(rl)` to return `result, err := Process(rl)` to avoid overloading of return err as `result` and `error` ? If result is returned, it is always added to the `rl` and returned error is used to indicate overall success or failure.

Reply:
Yes, I agree that having `Process` return both `result` and `error` will simplify the `AsMain` code. We would have to change this in the definition of `ResourceListProcessor` and make sure other functions implementing `Process` make the same change.

@mengqiy had a brief discussion and he brought up that the KRM function spec indicates that when a function fails, it should exit with a nonzero exit code. That means if Process runs into an error here, that error should be returned. 

That would mean that `Process` should record the error in `result`, and then return `result, err`.  i.e. `return &fn.Result{Message: err.Error(), Severity: fn.Error}, err`.

Then in `AsMain`, we can just append the result, do the yaml processing, and return the err from the function.

Reply:
Totally agree with the approach described above. Since, we only have two functions written using new SDK, changing the `ResourceListProcessor` definition shouldn't be a big deal, so we should go ahead and do it.

Reply:
I pushed a commit that pins this PR to my PR in the SDK https://github.com/GoogleContainerTools/kpt-functions-sdk/pull/534, which makes `Process` return both Result and error, so we can see what it will look like. 

We should wait for https://github.com/GoogleContainerTools/kpt-functions-sdk/pull/534 to go in, before this one (and remove the pinning to my fork).